.pre-commit-config.yaml

```diff
@@ -15,11 +15,11 @@ repos:
     rev: v2.2.2
     hooks:
     -   id: codespell
-        args: ["--count --ignore-words-list=ans,deriver,inout,packag --skip=\"*.js,*WordLists.swift,.git,Carthage,.build,build\""]
+        args: [--count --ignore-words-list="ans,deriver,inout,packag" --skip="*.js,*WordLists.swift,.git,Carthage,.build,build"]
 # SwiftLint is commented out until we fix all SwiftLint warnings and errors.
 # https://github.com/web3swift-team/web3swift/pull/756
-#-   repo: https://github.com/realm/SwiftLint
-#    rev: 0.50.3
-#    hooks:
-#    -   id: swiftlint
-#        args: [--fix, Sources, Tests]
+-   repo: https://github.com/realm/SwiftLint
+    rev: 0.50.3
+    hooks:
+    -   id: swiftlint
+        args: [Sources, Tests]  # TODO: Re-add `--fix`
```

Comment:
To be honest not sure about putting swiftlint into pre-commit and if it would be my choice, no way it should be there with `--fix` (auto fixing issues). In general, I think tools should rather hint at issues but how it's solved should be in the power of the developer. If there is too much stuff in pre-commits, it can get very annoying to work on the project - if not configured the right way.

Reply:
Swiftlint does not show errors that it fixes so with `--fix` I cannot see what we should temporarily disable.

Reply:
That's debatable in case of `swiftlint`. Though generally, I do agree that auto-fixes could be "scary". I've checked most of the rules and searched through `swiftlint` source code and out of 221 rules, out of which only 89 are enabled by default, only 67 rules support auto-fixes and out of them 42 are enabled by default. Most of them are about spacing and indents, or something similar.

`swiftlint --fix` will run either automatically when you commit something locally or manually if you get an error which must be fixed.

What I think will make sense to do is to replace `.pre-commit-config.yaml` with `sample.pre-commit-config.yaml`. README.md will hold a hint that `sample.pre-commit-config.yaml` should be copied and renamed to `.pre-commit-config.yaml` in order to enable `pre-commit` feature. While the `.pre-commit-config.yaml` will be gitignored allowing everyone to configure this file locally as they wish.

Reply:
> Swiftlint does not show errors that it fixes so with `--fix` I cannot see what we should temporarily disable.

Yea my comment was in general not about you disabling it temporarily.

Reply:
Just for the record, I'm +1 to disable forcing --fix feature on the swiftlint.

I see this as the different responsibility scopes, the one is comes within PR, where we are in charge to set our rules to the PR whatever we want and automatically deny any PR that violates them. 

But the other is the contributor development process before the PR, the local one, that could be and should be only up to a developer itself. So if they used to let's say implementing the whole with very bad code style within the process and then fixes it at the very end -- it should be allowed by default, and we shouldn't add a mess to them there.

As for me the warning thing is even discussable one, as this one could be considered as interruption into their internal scope. But that's not so critical to argue it without any negative feedback from a contributor folks beforehand.